src/unistd.rs

```diff
@@ -447,7 +447,14 @@ pub fn daemon(nochdir: bool, noclose: bool) -> Result<()> {
     Errno::result(res).map(drop)
 }
 
-pub fn sethostname(name: &[u8]) -> Result<()> {
+/// Set the system host name (see
+/// [gethostname(2)](http://man7.org/linux/man-pages/man2/gethostname.2.html)).
+///
+/// Given a name, attempt to update the system host name to the given string.
+/// On some systems, the host name is limited to as few as 64 bytes.  An error
+/// will be return if the name is not valid or the current process does not have
+/// permissions to update the host name.
+pub fn sethostname(name: &str) -> Result<()> {
```

Comment:
I'm not convinced `&str` is the right thing here. From your commit message:

> Although the underlying C API does take a pointer to a set of characters, *it is a requirement of almost every operating system that these bytes __not contain a premature NUL character__ or __other
special characters__*.  In other words, you want a `&str`.  Changing this to make the API make a bit more sense.

`&str` can contain a NUL byte anywhere, and is UTF-8, which may contain "special characters". Not really sure what to do here though!

PS commit message subject says `gethostname` when it should be `sethostname`.